pkg/net/net.go

```diff
@@ -151,3 +151,28 @@ type BroadcastChannel interface {
 // public key as its argument and returns true if the message should be
 // processed or false otherwise.
 type BroadcastChannelFilter func(*ecdsa.PublicKey) bool
+
+// Firewall represents a set of rules the remote peer has to conform to so that
+// a connection with that peer can be established. The rules are checked when
+// a new connection is established and periodically for all active connections.
+// If some of the expectations defined by firewall rules are no longer met, the
+// connection is dropped.
+type Firewall interface {
+
+	// Validate takes the remote peer public key and executes all the checks
+	// needed to decide whether the connection with the remote peer should be
+	// established and maintained.
+	// If expectations are not met, this function should return an error
+	// describing what is wrong.
+	Validate(remotePeerPublicKey *ecdsa.PublicKey) error
+}
```

Comment:
I think we are going to far with the description here. All of the details about when firewall rules are checked and what are the final effects are implementation details and I think they should be in places where `net.Firewall` is used (`authenticated_connection.go`, `watchtower.go`). Here we can limit ourselves to:

```
// Firewall represents a set of rules the remote peer has to conform to so that
// a connection with that peer can be considered valid.
type Firewall interface {

	// Validate takes the remote peer public key and executes all the checks.
	// If expectations are not met, this function should return an error
	// describing what is wrong.
	Validate(remotePeerPublicKey *ecdsa.PublicKey) error
}
```

What do you think?

Reply:
Done in https://github.com/keep-network/keep-core/pull/1612/commits/4838270d696b8dc87985f39fe9ea19ab5e91eae9, please let me know if you like it.